app/controllers/list_items_controller.rb

```diff
@@ -85,10 +85,28 @@ def confirm_destroy
     @list_item = @list.list_items.find(params[:id])
   end
 
+  def move
+    @list_item = @list.list_items.find(params[:id])
+  end
+
+  def update_move
+    @list_item = @list.list_items.find(params[:id])
+    @list_item.errors.add(:new_list_id, "Choose a list") if move_list_item_params.blank?
+    render :move and return if @list_item.errors.present?
+
+    new_list = @tag.lists.find(move_list_item_params)
+    new_index = new_list.list_items.map(&:index).max.present? ? new_list.list_items.map(&:index).max + 1 : 1
```

Comment:
What about:

```suggestion
    new_index = new_list.list_items.map(&:index).max.to_i + 1
```

If I'm getting this right, it'll only be not `present?` if the list is empty or `index` returned nil, i.e. in either of those cases `max` is nil. `nil.to_i` is 0, so we'd want to add 1 in either case; this way we can avoid recalculating the map and max.